### PR TITLE
Implements W3DFontLibrary.

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

File: src/game/client/gui/w3dgamefont.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        captainslog_error("W3D load font: unable to find font '%s' from asset manager", font->m_nameString.Str());
+        captainslog_dbgassert(0, "Missing or Corrupted Font.  Please see log for details");
+        return false;
+    }
+
+    font->m_fontData = font_chars;
+
+    FontCharsClass *alt_font_chars;
+
+    font->m_height = font_chars->Get_Char_Height();
```
Comment (xezon) [inline src/game/client/gui/w3dgamefont.cpp:43]: Style: Can move this behind `font->m_fontData = font_chars;`

```c++
font->m_fontData = font_chars;
font->m_height = font_chars->Get_Char_Height();
```

**Review by xezon — COMMENTED** [review]
I tried testing Fonts in Chinese language, but there is a infinite loop hit in

```
>	thyme.dll!FontCharsClass::Get_Char_Data(wchar_t ch) Line 75	C++
 	thyme.dll!FontCharsClass::Get_Char_Spacing(wchar_t ch) Line 106	C++
 	thyme.dll!Render2DSentenceClass::Allocate_New_Surface(const wchar_t * text, bool reuse_surface) Line 688	C++
 	thyme.dll!Render2DSentenceClass::Build_Sentence_Not_Centered(const wchar_t * text, int * x, int * y, bool reuse_surface) Line 945	C++
 	thyme.dll!Render2DSentenceClass::Get_Formatted_Text_Extents(const wchar_t * text) Line 502	C++
 	game.dat!007d62d6()	Unknown
 	game.dat![Frames below may be incorrect and/or missing, no symbols loaded for game.dat]	Unknown
 	game.dat!004f901e()	Unknown
 	game.dat!007cfcfe()	Unknown
 	game.dat!004fd46a()	Unknown
 	game.dat!004fc9b7()	Unknown
 	game.dat!006deb8c()	Unknown
 	game.dat!006dd9fd()	Unknown
 	game.dat!006dd455()	Unknown
 	game.dat!006d9c00()	Unknown
 	thyme.dll!RAMFile::Scan_String(Utf8String & string) Line 212	C++
 	00ee2574()	Unknown
 	game.dat!006dd35f()	Unknown
 	thyme.dll!RAMFile::Scan_String(Utf8String & string) Line 212	C++
 	00ee2574()	Unknown
 	game.dat!004f868d()	Unknown
 	game.dat!006dbd98()	Unknown
 	game.dat!0045d19c()	Unknown
```

Looks like `m_alternateUnicodeFont->m_alternateUnicodeFont` points to itself.

Not exactly related to this change, but this needs fixing to test font in Chinese.

debug2

**Comment by xezon** [comment]
For reference, Chinese packs here:
http://www.gentool.net/download/languages/

**Comment by tomsons26** [comment]
confirmed code has preexisting issues with asian fonts before this pr, functions in question are out of scope of this pr

**Pull request merged**